### PR TITLE
Split CI testing based on flutter channels

### PR DESCRIPTION
**Pull request by domesticmouse**
Split out the CI testing between `stable`, `beta`, and `dev`. This enables us to specify which samples are tested on which specific channel(s).

Note, the prior Github Actions CI configuration was masking various breaks, both on untested beta and dev targeted samples, and on the Android compilation CI. I've raised tracking bugs for the various breakages.

### CODE REVIEW

**Comment by domesticmouse** [comment]
PTAL @RedBrogdon 😄

**Comment by domesticmouse** [comment]
The android failure should be handled by https://github.com/flutter/samples/pull/669

The dev failures should be handled by https://github.com/flutter/samples/pull/668

**Comment by RedBrogdon** [comment]
@domesticmouse, this is great. I love that it'll get some of the experimental samples under test with the correct branches.

We had two PRs relating to CI setup in flight at the same time, and the other one has been landed. Mind doing a rebase and double-checking that everything is still how you want it?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -9,19 +9,19 @@ on:
   schedule:
     - cron: '0 0 * * *' # Every day at midnight
 
+defaults:
+  run:
+    shell: bash
```
Comment (RedBrogdon) [inline .github/workflows/main.yml:14]: Interesting. Is it running zsh on macOS?

Comment (RedBrogdon) [reply .github/workflows/main.yml:14]: Or PowerShell on Windows? 😄

**Review by RedBrogdon — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (domesticmouse) [reply .github/workflows/main.yml:14]: Powershell on windows by default was the issue.

Comment (RedBrogdon) [reply .github/workflows/main.yml:14]: Holy cow. I was just kidding around. 😄

Comment (domesticmouse) [reply .github/workflows/main.yml:14]: heh, holy cow is usual state of play for DevOps =)

**Pull request merged**